packages/vm/src/middleware/deterministic.rs

```diff
@@ -161,16 +161,17 @@ fn parse_wasm_opcode(opcode: &Operator) -> Result<(), CompileError> {
     }
 }
 
+/// Middleware is only supported in singlepass backend, see
+/// https://github.com/CosmWasm/cosmwasm/issues/311
 #[cfg(all(test, feature = "default-singlepass"))]
 mod tests {
-    use super::*;
-    use wasmer_runtime_core::{imports, typed_func::Func};
-
+    // No 'use super::*;' here. This is strange and means we are not testing the functions in this module directly.
```

Comment:
yeah, i didn't know how to test them unless they were in a stack.

i guess you could set up a `compile` stack with singlepass and deterministic middleware (but no metering) and use that in the place of `crate::backends::compile`

Reply:
Yeah, not a priority. But worth looking into at some point